src/d/actor/d_a_msw.cpp

```diff
@@ -4,52 +4,342 @@
+
+    s16 z = pos1.z * ((g_regHIO.mChild[0].mFloatRegs[0] + 20.0f) / i_ac->scale.z);
+    s16 x = -pos1.x * ((g_regHIO.mChild[0].mFloatRegs[0] + 20.0f) / i_ac->scale.x);
+
+    cLib_addCalcAngleS2(&i_ac->current.angle.x, z, 10, 0x800);
+    cLib_addCalcAngleS2(&i_ac->current.angle.z, x, 10, 0x800);
+
+    f32 dist = (g_regHIO.mChild[0].mFloatRegs[4] + 50.0f) * fabsf(pos1.z - pos2.z);
+    if (pActor->m2BC.x < dist) {
+        pActor->m2BC.x = dist;
+    }
+
+    dist = (g_regHIO.mChild[0].mFloatRegs[4] + 50.0f) * fabsf(pos1.x - pos2.x);
+    if (pActor->m2BC.z < dist) {
+        pActor->m2BC.z = dist;
+    }
+
+    dist = (g_regHIO.mChild[0].mFloatRegs[8] + 5.0f) * fabsf(pos1.x - pos2.x);
+    if (dist > 10.0f && pActor->m2B0.x < dist) {
+        cLib_addCalc2(&pActor->m2B0.x, dist, 1.0f, g_regHIO.mChild[0].mFloatRegs[7] + 1.2f);
+    }
+
+    dist = (g_regHIO.mChild[0].mFloatRegs[8] + 5.0f) * fabsf(pos1.z - pos2.z);
+    if (dist > 10.0f && pActor->m2B0.z < dist) {
+        cLib_addCalc2(&pActor->m2B0.z, dist, 1.0f, g_regHIO.mChild[0].mFloatRegs[7] + 1.2f);
+    }
+
 }
 
 /* 00000310-00000540       .text chain_Draw__FP9msw_class */
-void chain_Draw(msw_class*) {
-    /* Nonmatching */
+void chain_Draw(msw_class* i_this) {
+    s16 angle1;
+    s32 angle2;
+
+    for (int i = 0; i < 4; i++) {
+        cXyz tmp = i_this->m310[i] - i_this->m2E0[i];
+        angle1 = -cM_atan2s(tmp.y, tmp.z);
+
+        // TODO: std::sqrtf(float)
+        angle2 = cM_atan2s(tmp.x, sqrtf(tmp.y * tmp.y + tmp.z * tmp.z));
+
+        MtxTrans(i_this->m2E0[i].x, i_this->m2E0[i].y, i_this->m2E0[i].z, false);
+
+        s16 var_r4;
+        s32 var_r20;
+
+        if (i_this->m83C[i] != 0) {
+            var_r20 = i_this->m83C[i] * cM_ssin(i_this->m298 * 18000) * (g_regHIO.mChild[6].mFloatRegs[6] + 100.0f);
+            var_r4 = i_this->m298 * 3000;
+        } else {
+            var_r4 = 0;
+            var_r20 = 0;
+        }
+
+        cMtx_YrotM(*calc_mtx, var_r4);
+        cMtx_XrotM(*calc_mtx, angle1 + var_r20);
+        cMtx_YrotM(*calc_mtx, angle2);
+
+        i_this->mpChainModels[i]->setBaseTRMtx(*calc_mtx);
+        g_env_light.setLightTevColorType(i_this->mpChainModels[i], &i_this->tevStr);
+        mDoExt_modelUpdateDL(i_this->mpChainModels[i]);
+    }
 }
 
 /* 00000540-000005E8       .text daMsw_Draw__FP9msw_class */
-static BOOL daMsw_Draw(msw_class*) {
-    /* Nonmatching */
+static BOOL daMsw_Draw(msw_class* i_this) {
+    g_env_light.settingTevStruct(TEV_TYPE_BG0, &i_this->current.pos, &i_this->tevStr);
+    g_env_light.setLightTevColorType(i_this->mpModel, &i_this->tevStr);
+
+    mDoExt_modelUpdateDL(i_this->mpModel);
+
+    dComIfGd_setListBG();
+    chain_Draw(i_this);
+    dComIfGd_setList();
+
+    return TRUE;
 }
 
 /* 000005E8-0000080C       .text msw_move__FP9msw_class */
-void msw_move(msw_class*) {
-    /* Nonmatching */
+void msw_move(msw_class* i_this) {
+    i_this->m298 += 1;
+
+    // WTH?
+    // Branchless comparison
```

Comment:
Probably something #ifdef'd out for debug

Reply:
other than a switch statement, another way to get MWCC to generate a branchless comparison is with this trick using a self-assign on any float member variable that gets optimized out:
```cpp
    if (i_this->m29A == 0) {
        i_this->speedF = i_this->speedF;
    }
```
but without knowing what i_this->m29A there's no way to know what they actually wrote